### PR TITLE
HomePage View

### PR DESCRIPTION
**Pull request by ZachBerman**
-Home page for StoicChess
-Note changes to application_helper.rb to allow for different routes through devise
-Re-factored some rspec tests - Capybara does not like identical elements
-Reinstalled SimpleForm using bootstrap option

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -1,12 +1,112 @@
...
+  font-family: 'Balthazar', serif;
+}
+
+.clr {
+  clear: both;
+}
+
+/*FOOTER*/
+
+.Site {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/Site/site

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -1,12 +1,112 @@
...
+  display: flex;
+  min-height: 100vh;
+  flex-direction: column;
+}
+
+.Site-content {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/Site-content/site-content

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -1,12 +1,112 @@
...
+  flex: 1;
+}
+
+footer {
+  background-color: #979A9C;
+  height: 70px;
+}
+
+.footer_links li {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/footer_links/footer-links

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -1,12 +1,112 @@
...
+  list-style-type: none;
+}
+
+.footer_links li a {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/footer_links/footer-links

File: app/assets/stylesheets/application.scss [hunk 5]
```diff
@@ -1,12 +1,112 @@
...
+
+.remember {
+  padding-left: 10px;
+}
+
+.form-inputs input {
+  width: 100%;
+}
+
+.git_icon {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/git_icon/git-icon

File: app/assets/stylesheets/application.scss [hunk 6]
```diff
@@ -1,12 +1,112 @@
...
+  padding-left: 15%;
+}
+
+.link_icon {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/link_icon/link-icon

File: app/assets/stylesheets/application.scss [hunk 7]
```diff
@@ -1,12 +1,112 @@
...
+  padding-left: 30%;
+}
+
+.team_profile {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: Not a big deal, but FYI this breaks CSS naming conventions. 
s/team_profile/team-profile

File: app/assets/stylesheets/application.scss [hunk 8]
```diff
@@ -1,12 +1,112 @@
...
+  border-radius: 5px;
+  margin-top: 10px;
+}
+
+.feature_pic {
```
Comment (lasley) [inline app/assets/stylesheets/application.scss]: s/feature_pic/feature-pic

**Comment by ZachBerman** [comment]
so no underscores and capital letters?

File: app/views/games/index.html.erb [hunk 9]
```diff
@@ -1 +1,165 @@
...
+<br />
+<section id="register">
+  <div class="col-sm-5 col-sm-offset-1">
+    <%= image_tag 'chess.jpg', class: 'img-responsive feature_pic' %>
+    <br />
+  </div>
+  <div class="col-sm-3 col-sm-offset-1">
+    <%= simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+        <div class="form-inputs">
+          <br />
```
Comment (lasley) [inline app/views/games/index.html.erb]: `<br />` isn't really considered good practice IMO. You should be able to use Bootstrap in order to make this happen, but if you really need spacing you could use something like these in CSS (these are my gotos for a base spacing model):
```
.mt32 {
  margin-top: 32px !important;
}
.mt16 {
  margin-top: 16px !important;
}
.mt8 {
  margin-top: 8px !important;
}

.mb32 {
  margin-top: 32px !important;
}
.mb16 {
  margin-bottom: 16px !important;
}
.mb8 {
  margin-bottom: 8px !important;
}
```

File: app/views/games/index.html.erb [hunk 10]
```diff
@@ -1 +1,165 @@
...
+          <%= f.input :email, required: false, autofocus: true %>
+          <br />
+          <%= f.input :password, required: false %>
+          <br />
+          </div>
+          <%= f.input :remember_me, as: :boolean if devise_mapping.rememberable? %>
+          <%= render "devise/shared/forgotpassword" %>
+          <br />
+        <div class="form-inputs">
+          <%= f.button :submit, "LOG IN", class: 'btn btn-success login' %>
```
Comment (lasley) [inline app/views/games/index.html.erb]: This text conflicts with your other buttons, which seem to be title case. I would say s/LOG IN/Log In

**Comment by lasley** [comment]
Yeah CSS is basically snake case with dashes instead of underscores. There are reasons to use other things, typically around JS. This helps keep the namespaces clear.

File: app/views/games/index.html.erb [hunk 11]
```diff
@@ -1 +1,165 @@
...
+<section id="features">
+  <div class="col-sm-offset-2 col-sm-8 text-center">
+    <h2>CHESS APP - FINAL PROJECT</h2>
+    <h4>Description of the project and fact that it is our final project for FHP</h4>
+    <br />
+  </div>
+  <div class="col-sm-offset-2 col-sm-2 col-xs-offset-2 col-xs-8 text-center"/>
+    <%= image_tag 'chess.jpg', class: 'img-responsive feature_pic' %>
+    <h3>Feature 1</h3>
+    <p>Words words words, words words. Words words words, words words. Words words words, words words. Words words words, words words.</p>
```
Comment (lasley) [inline app/views/games/index.html.erb]: http://www.lipsum.com/ would be better than words words words

File: app/views/games/index.html.erb [hunk 12]
```diff
@@ -1 +1,165 @@
...
+  </div>
+</section>
+<br class="clr" />
+<hr class="col-sm-offset-1 col-sm-10"/>
+<br />
+<section id="features">
+  <div class="col-sm-offset-2 col-sm-8 text-center">
+    <h2>CHESS APP - FINAL PROJECT</h2>
+    <h4>Description of the project and fact that it is our final project for FHP</h4>
+    <br />
```
Comment (lasley) [inline app/views/games/index.html.erb]: Again may want to look at ways to remove the `<br />`. Hint: row and col `<div>`s & `<p>` tags are your friend

**Comment by ZachBerman** [comment]
Ok I added a WIP tag and will fix these things up.

File: app/views/games/index.html.erb [hunk 13]
```diff
@@ -1 +1,165 @@
...
+          <%= render "devise/shared/forgotpassword" %>
+          <br />
+        <div class="form-inputs">
+          <%= f.button :submit, "LOG IN", class: 'btn btn-success login' %>
+        </div>
+    <% end %>
+  </div>
+</section>
+<br class="clr" />
+<hr class="col-sm-offset-1 col-sm-10"/>
```
Comment (lasley) [inline app/views/games/index.html.erb]: You shouldn't use columns without rows. This breaks the nesting and will flow all of your columns into each other

File: app/views/games/index.html.erb [hunk 14]
```diff
@@ -1 +1,165 @@
...
+<br />
+<section id="features">
+  <div class="col-sm-offset-2 col-sm-8 text-center">
```
Comment (lasley) [inline app/views/games/index.html.erb]: See above re cols w/o rows

File: app/views/games/index.html.erb [hunk 15]
```diff
@@ -1 +1,165 @@
...
+    <%= image_tag 'chess.jpg', class: 'img-responsive feature_pic' %>
+    <h3>Feature 2</h3>
+    <p>Words words words, words words. Words words words, words words. Words words words, words words. Words words words, words words.</p>
+  </div>
+  <div class="col-sm-offset-1 col-sm-2 col-xs-offset-2 col-xs-8 text-center"/>
+    <%= image_tag 'chess.jpg', class: 'img-responsive feature_pic' %>
+    <h3>Feature 3</h3>
+    <p>Words words words, words words. Words words words, words words. Words words words, words words. Words words words, words words.</p>
+  </div>
+  <br class="clr" />
```
Comment (lasley) [inline app/views/games/index.html.erb]: Same thing with the brs, see where you can eliminate. There are only some very rare instances in which breaks are considered good practice. Think of them like tables - easy but inconsistent between devices.

File: app/views/games/index.html.erb [hunk 16]
```diff
@@ -1 +1,165 @@
...
+<hr class="col-sm-offset-1 col-sm-10"/>
+<section id="team">
+  <div class="text-center">
+    <h2>THE TEAM</h2>
+    <br class="clr" />
+  </div>
+  <div class="col-sm-offset-1 col-sm-2"/>
+    <%= image_tag 'chad.png', class: 'img-responsive team_profile' %>
+    <h5 class="text-center">Chad</h5>
+    <%= image_tag 'linked_in.png', class: 'link_icon' %>
```
Comment (lasley) [inline app/views/games/index.html.erb]: Bootstrap has built in http://glyphicons.com/ - I recommend using those for the social icons.

File: app/views/games/index.html.erb [hunk 17]
```diff
@@ -1 +1,165 @@
...
+    <%= link_to 'https://github.com/chadbaum' do %>
+      <%= image_tag 'github.png', class: 'git_icon' %>
+    <% end %>
+  </div>
+  <div class="col-sm-2 col-xs-8"/>
```
Comment (lasley) [inline app/views/games/index.html.erb]: Seems like a lot of repetition here. What if we added an array of hashes in the controller with the keys `name`, `linked_in`, `github`, etc. then just provide that to our template, loop, and output to page?

The assumption could easily be made that the last one in the array is yoda.

File: app/views/layouts/application.html.erb [hunk 18]
```diff
@@ -7,19 +7,59 @@
...
-    <li><%= link_to 'Sign out', destroy_user_session_path, method: :delete %></li>
-  <% else %>
-      <li><%= link_to 'Sign in', new_user_session_path %></li>
-      <li><%= link_to 'Sign up', new_user_registration_path %></li>
-  <% end %>
+          <!-- Collect the nav links, forms, and other content for toggling -->
+          <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+            <ul class="nav navbar-nav navbar-right">
+              <% if user_signed_in? %>
+                <li><%= link_to 'MY PROFILE', root_path %></li>
```
Comment (lasley) [inline app/views/layouts/application.html.erb]: Hmm maybe I was wrong on a previous comment regarding everything being title casing, but there is definitely inconsistency. I recommend picking one and sticking to it, probably title case.

File: app/views/layouts/application.html.erb [hunk 19]
```diff
@@ -7,19 +7,59 @@
...
+            <li><%= link_to 'SIGN IN', new_user_session_path %></li>
+            <li><%= link_to 'SIGN UP', new_user_registration_path %></li>
+          <% end %>
+        </ul>
+      </div>
+      <div class="col-xs-4 text-center">
+        <ul class="footer_links">
+          <li><%= link_to 'TEAM STOIC ON GITHUB', 'https://github.com/TeamStoic' %></li>
+          <li><%= link_to 'THE FIREHOSE PROJECT', 'https://www.thefirehoseproject.com/' %></li>
+          <li><%= link_to 'COPYRIGHT', '#' %></li>
```
Comment (lasley) [inline app/views/layouts/application.html.erb]: Link to `#`?

File: app/views/games/index.html.erb (hunk 9 above)
Comment (ZachBerman) [inline app/views/games/index.html.erb]: I thought using !important was bad practice?

Comment (lasley) [inline app/views/games/index.html.erb]: Yup definitely, but in this instance it's fine. Nuances are quite ridiculous, but for example we can see that Bootstrap 4 will contain basically the same code: http://v4-alpha.getbootstrap.com/components/utilities/#spacing

File: app/views/layouts/application.html.erb (hunk 19 above)
Comment (ZachBerman) [inline app/views/layouts/application.html.erb]: Just a placeholder for a copyright page if we choose to use it...

File: app/views/devise/sessions/new.html.erb [hunk 20]
```diff
@@ -1,14 +1,16 @@
 <h2>Log in</h2>
 
 <%= simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
-  <div class="form-inputs">
-    <%= f.input :email, required: false, autofocus: true %>
-    <%= f.input :password, required: false %>
-    <%= f.input :remember_me, as: :boolean if devise_mapping.rememberable? %>
-  </div>
+  <div class="deviseNewLogIn">
```
Comment (lasley) [inline app/views/devise/sessions/new.html.erb]: CSS classes are typically lowercase with hyphenated joins `devise-new-login`

**Comment by lasley** [comment]
👍

**Comment by Se7enB2st** [comment]
👍

**Pull request merged**